File: Kernel Simulator/Shell/Shells/FTP/Commands/GetFolder.vb
﻿
'    Kernel Simulator  Copyright (C) 2018-2022  Aptivi
'
'    This file is part of Kernel Simulator
'
'    Kernel Simulator is free software: you can redistribute it and/or modify
'    it under the terms of the GNU General Public License as published by
'    the Free Software Foundation, either version 3 of the License, or
'    (at your option) any later version.
'
'    Kernel Simulator is distributed in the hope that it will be useful,
'    but WITHOUT ANY WARRANTY; without even the implied warranty of
'    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
'    GNU General Public License for more details.
'
'    You should have received a copy of the GNU General Public License
'    along with this program.  If not, see <https://www.gnu.org/licenses/>.

Imports KS.Network.FTP.Transfer

Namespace Shell.Shells.FTP.Commands
    ''' <summary>
    ''' Downloads a folder from the current working directory
    ''' </summary>
    ''' <remarks>
    ''' Downloads the binary or text folder and saves it to the current working local directory for you to use the downloaded folder that is provided in the FTP server.
    ''' </remarks>
    Class FTP_GetFolderCommand
        Inherits CommandExecutor
        Implements ICommand

        Public Overrides Sub Execute(StringArgs As String, ListArgs() As String, ListArgsOnly As String(), ListSwitchesOnly As String()) Implements ICommand.Execute
            Dim RemoteFolder As String = ListArgs(0)
            Dim LocalFolder As String = If(ListArgs.Count > 1, ListArgs(1), "")
            Write(DoTranslation("Downloading folder {0}..."), True, ColTypes.Progress, RemoteFolder)
            Dim Result As Boolean = If(Not String.IsNullOrWhiteSpace(LocalFolder), FTPGetFolder(RemoteFolder, LocalFolder), FTPGetFolder(RemoteFolder))
            If Result Then
                Console.WriteLine()
                Write(DoTranslation("Downloaded folder {0}."), True, ColTypes.Success, RemoteFolder)
            Else
                Console.WriteLine()
                Write(DoTranslation("Download failed for folder {0}."), True, ColTypes.Error, RemoteFolder)
            End If
        End Sub

    End Class
End Namespace

File: Kernel Simulator/Network/FTP/Commands/GetFolder.cs
﻿using System.Linq;
using KS.ConsoleBase.Colors;
using KS.Languages;
using KS.Misc.Writers.ConsoleWriters;

// Kernel Simulator  Copyright (C) 2018-2022  Aptivi
// 
// This file is part of Kernel Simulator
// 
// Kernel Simulator is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
// 
// Kernel Simulator is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
// 
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.

using KS.Network.FTP.Transfer;
using KS.Shell.ShellBase.Commands;

namespace KS.Network.FTP.Commands
{
	class FTP_GetFolderCommand : CommandExecutor, ICommand
	{

		public override void Execute(string StringArgs, string[] ListArgs, string[] ListArgsOnly, string[] ListSwitchesOnly)
		{
			string RemoteFolder = ListArgs[0];
			string LocalFolder = ListArgs.Count() > 1 ? ListArgs[1] : "";
			TextWriterColor.Write(Translate.DoTranslation("Downloading folder {0}..."), true, color: KernelColorTools.GetConsoleColor(KernelColorTools.ColTypes.Progress), RemoteFolder);
			bool Result = !string.IsNullOrWhiteSpace(LocalFolder) ? FTPTransfer.FTPGetFolder(RemoteFolder, LocalFolder) : FTPTransfer.FTPGetFolder(RemoteFolder);
			if (Result)
			{
				TextWriterColor.WritePlain("", true);
				TextWriterColor.Write(Translate.DoTranslation("Downloaded folder {0}."), true, color: KernelColorTools.GetConsoleColor(KernelColorTools.ColTypes.Success), RemoteFolder);
			}
			else
			{
				TextWriterColor.WritePlain("", true);
				TextWriterColor.Write(Translate.DoTranslation("Download failed for folder {0}."), true, color: KernelColorTools.GetConsoleColor(KernelColorTools.ColTypes.Error), RemoteFolder);
			}
		}

	}
}